contrib/exporters/awsflowlogs/mod/mangle_logstatus.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"time"
+
+	cache "github.com/pmylund/go-cache"
+	"github.com/spf13/viper"
+
+	"github.com/skydive-project/skydive/common"
+	"github.com/skydive-project/skydive/flow"
+)
+
+type mangleLogStatus struct {
+	linkIDs map[string]true
+	flows   map[string]*flow.Flow
+}
+
+// Mangle log-status book keeping
+func (m *mangleLogStatus) Mangle(in []*flow.Flow) (out []*flow.Flow) {
+	now := common.UnixMillis(time.Now())
+	// OK: all flows
+	for _, flow := range in {
+		okFlow := *flow
+		okFlow.LogStatus = FlowLogStatus_OK
+		out = append(out, &okFlow)
+	}
+
+	// NODATA: when nothing was seen per intrerface in measurement window
+	linkIDs := make(map[string]true)
+	for _, flow := range in {
+		linkIDs[flow.Link.ID] = true
+	}
+
+	for key := range linkIDs {
+		if !m.linkIDs[key] {
+			noFlow := &flow.Flow{
+				Link: &FlowLayer{
+					ID: key,
+				},
+				Start:     now,
+				Last:      now,
+				LogStatus: FlowLogStatus_NODATA,
+			}
+			out = append(out, noFlow)
+		}
+
+	}
+
+	for _, flow := range in {
+		m.linkIDs[flow.Link.ID] = true
```

Comment:
linkIDs --> should be IDs NOT Bool ... --> change the name of the map to represent boolean

Reply:
@eranra - what name exactly are you proposing?

Reply:
will change to `linkIdExists`